contracts/src/interfaces/IStakeTable.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: Unlicensed
+
+pragma solidity ^0.8.0;
+
+import { BN254 } from "bn254/BN254.sol";
+import { EdOnBN254 } from "../libraries/EdOnBn254.sol";
+
+/// @title Interface for stake table that keep track of validators' external key and staked amount.
+/// @dev Stake delegation happens in a separate DelegationPool contract, and specific to
+/// instantiation thus not part of this interface
+interface IStakeTable {
+    /// @notice Supported stake type, either using native token or re-staked using ETH
+    enum StakeType {
+        Native,
+        Restake
+    }
+
+    /// @dev (sadly, Solidity doesn't support type alias on non-primitive types)
```

Comment:
New feature! https://docs.soliditylang.org/en/latest/types.html#user-defined-value-types

Reply:
Yes I am aware of this, it doesn't support **non-primitive/built-in underlying types tho**:

> A user-defined value type is defined using type C is V, where C is the name of the newly introduced type and V has to be a built-in value type (the “underlying type”).

Reply:
Oh that's so annoying lol